src/OpenLoco/Title.cpp

```diff
@@ -81,13 +161,73 @@ namespace OpenLoco::Title
         initialiseViewports();
         sub_4284C8();
         Gui::init();
-        sub_444357();
+        reset();
         Gfx::invalidateScreen();
         resetScreenAge();
 
         Audio::playTitleScreenMusic();
     }
 
+    // 0x00444387
+    void update()
+    {
+        if (!isTitleMode())
+            return;
+
+        resetScreenAge();
+
+        if (_waitCounter > 0)
+        {
+            _waitCounter -= 1;
+            return;
+        }
+
+        do
+        {
+            auto& command = *_sequenceIterator++;
+            std::visit(overloaded{
+                           [](WaitStep step) {
+                               _waitCounter = step.duration / 4;
+                           },
+                           [](ReloadStep step) {
+                               loadTitle();
+                               Gfx::invalidateScreen();
+                               resetScreenAge();
+                               addr<0x50C19A, uint16_t>() = 55000;
+                           },
+                           [](MoveStep step) {
+                               if (addr<0x00525E28, uint32_t>() & 1)
+                               {
+                                   auto pos = Map::map_pos(step) + Map::map_pos(16, 16);
+                                   auto height = Map::TileManager::getHeight(pos);
+                                   auto main = Ui::WindowManager::getMainWindow();
+                                   if (main != nullptr)
+                                   {
+                                       auto pos3d = Map::map_pos3(pos.x, pos.y, height.landHeight);
+                                       main->viewportCentreOnTile(pos3d);
+                                       main->flags &= ~Ui::WindowFlags::scrolling_to_location;
+                                       main->viewportsUpdatePosition();
+                                   }
+                               }
+                           },
+                           [](RotateStep step) {
+                               if (addr<0x00525E28, uint32_t>() & 1)
+                               {
+                                   auto main = Ui::WindowManager::getMainWindow();
+                                   if (main != nullptr)
+                                   {
+                                       main->viewportRotateRight();
+                                   }
+                               }
+                           },
+                           [](ResetStep step) {
+                               _sequenceIterator = _titleSequence.begin();
+                           },
+                       },
+                       command);
+        } while (_waitCounter == 0);
+    }
```

Comment:
the original did `_waitCounter--;` just before the return.

Reply:
Hmm, That would get it to -1, right? Aside from processing a WaitStep, I only see it decrementing. That worries me; it might never leave the loop in the next iteration.

If we do go this route, we've currently declared `_waitCounter` a `uint16_t`, so it would need to be changed to `int16_t`.

Reply:
No it wouldn't the loop is looking for a non 0 value for wait counter. So we know at the end of the loop that it is non zero so a decrement will be fine.

Reply:
Tbh, I think it's cleaner to leave the code as-is. It makes no difference to the title screen running, and having `_waitCounter` be >= 0 at all times feels better to me.

Reply:
It just means that the sequence will be one tick longer. As long as you are fine with that.

Reply:
I would argue that the implementation is "more accurate"? The original would consume a tick for the current wait command, now that depends on your viewpoint for when the countdown should start.

Reply:
Depends if the sequence is meant to align with the music.

Reply:
I'm very much leaning towards dropping it and accepting the 1/40th difference. However, @ZehMatt's suggestion below looks like a sensible compromise. Any thoughts about his suggestion, @duncanspumpkin?

Reply:
I don't care that much either way. Just wanted to make sure it's thought about and the implications of if it's not there